.vscode/settings.json

```diff
@@ -0,0 +1,34 @@
+{
+	"python.formatting.provider": "none",
+
+	"cssvar.files": ["./js/node_modules/pollen-css/pollen.css"],
+	// Do not ignore css files in node_modules, which is ignored by default
+	"cssvar.ignore": [],
+	// Use Pollen's inbuilt variable ordering
+	"cssvar.disableSort": true,
+	// Add support for autocomplete in other file types
+	"cssvar.extensions": ["js", "css", "html", "jsx", "tsx", "svelte"],
+	"python.analysis.extraPaths": ["./gradio/themes/utils"],
+	"prettier.useTabs": true,
+	"editor.codeActionsOnSave": {
+		"source.fixAll": true
+	},
+	"editor.formatOnSave": true,
+	"svelte.plugin.svelte.format.enable": true,
+	"prettier.configPath": ".config/.prettierrc.json",
+	"prettier.ignorePath": ".config/.prettierignore",
+	"python.testing.pytestArgs": ["."],
+	"python.testing.unittestEnabled": false,
+	"python.testing.pytestEnabled": true,
+	"eslint.packageManager": "pnpm",
+	"eslint.probe": ["javascript", "typescript", "html", "markdown", "svelte"],
+	"eslint.validate": ["javascript", "typescript", "html", "markdown", "svelte"],
+	"eslint.lintTask.options": ".",
+	"eslint.options": {
+		"overrideConfigFile": "./.config/.eslintrc.cjs",
+		"ignorePath": "./.config/.eslintignore"
+	},
+	"[python]": {
+		"editor.defaultFormatter": "ms-python.black-formatter"
+	}
+}
```

Comment:
These options are either there to make extensions work without contributors needing to loads of manual work or disable things to prevent conflicts. These aren't things anyone would want to switch off really, they are welcome to not use extensions but if they _do_ want to use them, not using these settings will cause them to fail. SO I don't think it is particularly restrictive.

I have reduced them a little by switching config formats but most of these are needed.

Reply:
(Most of these settings are just being reinstated from a while ago when they were incorrectly deleted).

Reply:
Just for context, it has taken me hours to get the correct combination of eslint + eslint extension settings to make it work, thats why i've added them. I don't think we want our contributors to go through that process.